test/e2e/container_iface_name_test.go

```diff
@@ -0,0 +1,277 @@
+		// macvlan/ipvlan network, created with a specific parent
+		// interface, continues to be ethX when interface_name in the
+		// containers.conf file is set to default value, i.e., "".
+
+		SkipIfRootless("cannot create network device in rootless mode.")
+
+		for _, driverType := range []string{"macvlan", "ipvlan"} {
+			if driverType == "ipvlan" && isNetavark(podmanTest) {
+				continue
+			}
+
+			// Create a nic to be used as a parent for macvlan/ipvlan network.
+			nicName1 := createNetworkName("nic")[:8]
+			nicName2 := createNetworkName("nic")[:8]
+
+			netName1 := createNetworkName(driverType)
+			netName2 := createNetworkName(driverType)
+
+			parent1 := "parent=" + nicName1
+			parent2 := "parent=" + nicName2
+
+			defer deleteNetworkDevice(nicName1)
+			createNetworkDevice(nicName1)
+
+			defer podmanTest.removeNetwork(netName1)
+			nc1 := podmanTest.Podman([]string{"network", "create", "-d", driverType, "-o", parent1, "--subnet", "10.10.0.0/24", netName1})
+			nc1.WaitWithDefaultTimeout()
+			Expect(nc1).Should(ExitCleanly())
+
+			ctr := podmanTest.Podman([]string{"run", "-d", "--network", netName1, "--name", "test", ALPINE, "top"})
+			ctr.WaitWithDefaultTimeout()
+			Expect(ctr).Should(ExitCleanly())
+
+			exec1 := podmanTest.Podman([]string{"exec", "test", "ip", "addr", "show", "eth0"})
+			exec1.WaitWithDefaultTimeout()
+			Expect(exec1).Should(ExitCleanly())
+			Expect(exec1.OutputToString()).Should(ContainSubstring("eth0"))
+
+			defer deleteNetworkDevice(nicName2)
+			createNetworkDevice(nicName2)
+
+			defer podmanTest.removeNetwork(netName2)
+			nc2 := podmanTest.Podman([]string{"network", "create", "-d", driverType, "-o", parent2, "--subnet", "10.25.40.0/24", netName2})
+			nc2.WaitWithDefaultTimeout()
+			Expect(nc2).Should(ExitCleanly())
+
+			conn := podmanTest.Podman([]string{"network", "connect", netName2, "test"})
+			conn.WaitWithDefaultTimeout()
+			Expect(conn).Should(ExitCleanly())
+
+			exec2 := podmanTest.Podman([]string{"exec", "test", "ip", "addr", "show", "eth1"})
+			exec2.WaitWithDefaultTimeout()
+			Expect(exec2).Should(ExitCleanly())
+			Expect(exec2.OutputToString()).Should(ContainSubstring("eth1"))
+
+			rm := podmanTest.Podman([]string{"rm", "--time=0", "-f", "test"})
+			rm.WaitWithDefaultTimeout()
+			Expect(rm).Should(ExitCleanly())
+		}
+	})
+
+	It("podman container interface name with device scheme for macvlan/ipvlan network with parent", func() {
+		// Assert that the network interface name inside container for
+		// macvlan/ipvlan network, created with a specific parent
+		// interface, is the parent interface name ethX when
+		// interface_name in the containers.conf file is set to "device"
+
+		SkipIfRootless("cannot create network device in rootless mode.")
+
+		createContainersConfFileWithDeviceIfaceName(podmanTest)
+
+		for _, driverType := range []string{"macvlan", "ipvlan"} {
+			if driverType == "ipvlan" && isNetavark(podmanTest) {
```

Comment:
These tests must work with netavark, since https://github.com/containers/podman/pull/21410 we will no longer have any CNI tests so this will never be run in CI.
What is the error reported?

Reply:
It failed with a very clear statement that netavark does not support ipvlan.